### PR TITLE
Create migration guide for RouteSettings.copywith

### PR DESCRIPTION
**Pull request by chunhtai**
needs to wait for pr to land https://github.com/flutter/flutter/pull/113860

_Description of what this PR is changing or adding, and why:_

_Issues fixed by this PR (if any):_ Fixes #ISSUE-NUMBER

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by chunhtai** [comment]
PR has landed, this migration guide is ready for review

File: src/release/breaking-changes/routesettings-copywith-migration.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+```dart
+RouteSettings newSettings = RouteSettings(name: 'new name', arguments: oldSettings.arguments);
+```
+
+
+## Timeline
+
+Landed in version: 3.5.0-9.0.pre-137-gc6f6095acd<br>
+In stable release: NEXT
```
Comment (chunhtai) [inline src/release/breaking-changes/routesettings-copywith-migration.md:37]: What should i put in here?

Comment (domesticmouse) [reply src/release/breaking-changes/routesettings-copywith-migration.md:37]: @sfshaza2 to confirm, but I believe we leave this as TBA until it actually ships in a stable release.

File: src/release/breaking-changes/routesettings-copywith-migration.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
+title: Migration guide for RouteSettings copyWith
+description: Removal of RouteSettings copyWith and how to migrate
+---
+
+## Summary
+
+RouteSettings copyWith was removed, and existing apps needed to use the constructor to create a new RouteSettings instead.
```
Comment (domesticmouse) [inline src/release/breaking-changes/routesettings-copywith-migration.md:8]: Consider wrapping code with backticks, e.g. `copyWith`

Comment (chunhtai) [reply src/release/breaking-changes/routesettings-copywith-migration.md]: done

File: src/release/breaking-changes/routesettings-copywith-migration.md (hunk 1 above)
Comment (chunhtai) [reply src/release/breaking-changes/routesettings-copywith-migration.md]: changed to TBA

**Comment by chunhtai** [comment]
a friendly bump

File: src/release/breaking-changes/routesettings-copywith-migration.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+---
+
+## Summary
+
+`RouteSettings.copyWith` was removed, and existing apps needed to use the constructor to create a new `RouteSettings` instead.
+
+## Context
+
+With the introduction of [`Page`][] class,
+the `RouteSettings.copyWith` was no longer a viable API.
```
Comment (sfshaza2) [inline src/release/breaking-changes/routesettings-copywith-migration.md:13]:
```suggestion
With the introduction of the [`Page`][] class,
the `RouteSettings.copyWith` was no longer a viable API.
```

File: src/release/breaking-changes/routesettings-copywith-migration.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+## Description of change
+
+`RouteSettings.copyWith` was removed
+
+## Migration guide
+
+Existing apps:
```
Comment (sfshaza2) [inline src/release/breaking-changes/routesettings-copywith-migration.md:21]:
```suggestion
Code before migration:
```

File: src/release/breaking-changes/routesettings-copywith-migration.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+```dart
+RouteSettings newSettings = oldSettings.copyWith(name: 'new name');
+```
+
+After migrations:
```
Comment (sfshaza2) [inline src/release/breaking-changes/routesettings-copywith-migration.md:27]:
```suggestion
Code after migration:
```

**Review by sfshaza2 — COMMENTED** [review]
A couple tweaks. Also, can you add this to the index file?

**Comment by chunhtai** [comment]
@sfshaza2 Thanks for review, added a item in index.md.

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Pull request merged**